akka-http2-support/src/main/scala/akka/http/javadsl/Http2.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.event.LoggingAdapter
+import akka.http.javadsl.model.{ HttpRequest, HttpResponse }
+import akka.http.javadsl.settings.ServerSettings
+import akka.http.impl.util.JavaMapping.Implicits._
+import akka.japi.Function
+import akka.stream.Materializer
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.Future
+
+object Http2 extends ExtensionId[Http2] with ExtensionIdProvider {
```

Comment:
If we're introducing it, then please mirror methods from scaladsl's Http2 or mark with huge FIXME's that it's not mirroring all methods yet